requests/cookies.py

```diff
@@ -57,6 +57,12 @@ def get_full_url(self):
         # If they did set it, retrieve it and reconstruct the expected domain
         host = self._r.headers['Host']
         parsed = urlparse(self._r.url)
+
+        # If parsed url is str type, ensure that host is also str type
+        if isinstance(parsed.scheme, str) and not isinstance(host, str)\
+                and isinstance(host, bytes):
+            host = host.decode('ascii')
```

Comment:
A quick dive into the CPython codebase shows that the current behaviour of the cookiejar module uses native strings (`str`). The same is true on Python 2.

That means that what we really want to do is just to call `to_native_str` on `host`. That is a Requests-specific function that lives in `utils`. That can skip a lot of this if statement.

While we're here, I should also note that `ascii` is probably a bad default encoding choice: we should at least consider defaulting to something a bit more sensible, like UTF-8.